rotkehlchen/chain/ethereum/names.py

```diff
@@ -67,56 +67,135 @@ def search_for_addresses_names(
+                return account.label
+
+    return ""
+
+
+def private_addressbook_address_to_name(db: DBHandler, address: ChecksumEvmAddress) -> str:
+    db_addressbook = DBAddressbook(db)
+    with db.conn.read_ctx() as cursor:
+        addressbook_entries = db_addressbook.get_addressbook_entries(
             cursor=cursor,
-            blockchain=SupportedBlockchain.ETHEREUM,
+            addresses=[address],
         )
+        for addressbook_entry in addressbook_entries:
+            if addressbook_entry.address == address:
+                return addressbook_entry.name
+
+    return ""
+
+
+def global_addressbook_address_to_name(db: DBHandler, address: ChecksumEvmAddress) -> str:
+    db_addressbook = DBAddressbook(db)
+    with db_addressbook.read_ctx(AddressbookType.GLOBAL) as global_cursor:
+        addressbook_entries = db_addressbook.get_addressbook_entries(
+            cursor=global_cursor,
+            addresses=[address],
+        )
+        for addressbook_entry in addressbook_entries:
+            if addressbook_entry.address == address:
+                return addressbook_entry.name
+
+    return ""
 
-    for account in accounts_data:
-        if account.address in addresses_set:
-            labels[account.address] = account.label
 
-    result_mappings: Dict[ChecksumEvmAddress, str] = {}
+def hardcoded_address_to_name(_: DBHandler, address: ChecksumEvmAddress) -> str:
+    constant_name = ETHADDRESS_TO_KNOWN_NAME.get(address, "")
+    return constant_name
+
+
+def token_mappings_address_to_name(_: DBHandler, address: ChecksumEvmAddress) -> str:
+    token_mappings = GlobalDBHandler().get_tokens_mappings(addresses=[address])
+
+    for _addr, name in token_mappings.items():
+        return name
+
+    return ""
+
+
+def ens_address_to_name(db: DBHandler, address: ChecksumEvmAddress) -> str:
+    db_ens = DBEns(db)
+
+    with db.conn.read_ctx() as cursor:
+        db_reverse_ens = db_ens.get_reverse_ens(cursor, addresses=[address])
+        reverse_ens = list(db_reverse_ens.values())
+
     for mapping in reverse_ens:
-        if isinstance(mapping, EnsMapping):
-            result_mappings[mapping.address] = mapping.name
-    for address, name in mapped_tokens.items():
-        result_mappings[address] = name
-    for hardcoded_address, hardcoded_name in hardcoded_mappings.items():
-        result_mappings[hardcoded_address] = hardcoded_name
-    for addressbook_entry in global_addressbook:
-        result_mappings[addressbook_entry.address] = addressbook_entry.name
-    for addr, label in labels.items():
-        if label is not None:
-            result_mappings[to_checksum_address(addr)] = label
-    for addressbook_entry in private_addressbook:
-        result_mappings[addressbook_entry.address] = addressbook_entry.name
-
-    return result_mappings
+        if isinstance(mapping, EnsMapping) and mapping.name:
+            return mapping.name
+
+    return ""
```

Comment:
What are you doing here? You are taking a mapping of 1 value. The value you need. Then you turn that into a list? Why? And then iterate that list? And check that the list's type is what the list has and return the first item?!?!

Just do this:

```python
db_ens = DBEns(db)
with db.conn.read_ctx() as cursor:
    db_reverse_ens = db_ens.get_reverse_ens(cursor, addresses=[address])
    return db_reverse_ens.get(address, None)
```

Reply:
As far as I understand, at least the `isinstance` check must be there before returning a value as the `get_reverse_ens` method may return a `Timestamp`

Reply:
Hmmm that's true. Then you can keep the is_instance there.

But then the `get_reverse_ens` is terribly designed by whoever made it.

We would need to change it to only return mapped names, and have a different named function that also returns timestamp. But that can be done in another PR.